### PR TITLE
Add more platform specific screen from Window/Rect/Point methods.

### LINKED ISSUE #7164: Wrong popup position when working with multiple monitors
**Describe the bug**
We're experiencing an issue positioning a context menu, when the Avalonia window is between two screens with different DPI configuration. Please, see the following video for a detailed explanation:

https://www.loom.com/share/e30aea24416b4209aacccdbb85e0c51d

**To Reproduce**
Minimal repro case repository: https://github.com/danipen/ContextMenuPlacementBug

1. Execute the example.
2. Maximize the window.
3. Click the button to show the context menu.

**Desktop**

- OS: Windows
- Version: 0.10.1

### LINKED ISSUE #7761: Child Window does not get CenterScreen when parent window is maximized
**Describe the bug**
When the parent window is maximized, the child window does not get centered. Event with the `WindowStartupLocation.CenterScreen` option. This issue happens on Windows OS only.

**To Reproduce**

1. Open Window;
2. Maximize it;
3. Open child window with the `WindowStartupLocation.CenterScreen` option:
4. **BUG**: Child Window does not open in the center of the screen;

**Expected behavior**
Expect child window to be shown in the center of the screen

**Screenshots**
Code:
image

Child window shown location:
image

**Desktop:**
- OS:  Windows only;
- Version [From 0.10.12]

### PR DESCRIPTION
**Pull request by MarchingCube**
## What does the pull request do?

PR expanding platform specific screen APIs to fix several problems with child window/dialog positioning.

## What is the current behavior?

One note about our WPF compatibility: https://docs.microsoft.com/en-us/dotnet/api/system.windows.windowstartuplocation?view=windowsdesktop-6.0#System_Windows_WindowStartupLocation_CenterScreen

In WPF CenterScreen is using cursor position for centering
In Avalonia it is based on the window.

## What is the updated/expected behavior with this PR?

Correct platform specific behavior when requesting screen info.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Fixed issues

Fixes #7761 
Fixes #7164

### CODE REVIEW

File: src/Avalonia.Controls/Screens.cs [hunk 1]
```diff
@@ -20,30 +20,53 @@ public Screens(IScreenImpl iScreenImpl)
...
+        }
+
         public Screen ScreenFromPoint(PixelPoint point)
         {
-            return All.FirstOrDefault(x => x.Bounds.Contains(point));        
+            var screen = _iScreenImpl.ScreenFromPoint(point);
+
+            if (screen == null)
+            {
+                screen = All.FirstOrDefault(x => x.Bounds.Contains(point));
```
Comment (grokys) [inline src/Avalonia.Controls/Screens.cs:66]: As described [here](https://github.com/AvaloniaUI/Avalonia/issues/7164#issuecomment-994761322) `Rect.Contains` is inclusive on the right/bottom edge and so could give the wrong result here.

File: src/Avalonia.Controls/Platform/ScreenHelper.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+namespace Avalonia.Platform
+{
+    public static class ScreenHelper
+    {
+        public static Screen? ScreenFromPoint(PixelPoint point, IReadOnlyList<Screen> screens)
+        {
+            foreach (Screen screen in screens)
+            {
+                if (screen.Bounds.Contains(point))
```
Comment (grokys) [inline src/Avalonia.Controls/Platform/ScreenHelper.cs:14]: As described [here](https://github.com/AvaloniaUI/Avalonia/issues/7164#issuecomment-994761322) `Rect.Contains` is inclusive on the right/bottom edge and so could give the wrong result here.

Comment (grokys) [reply src/Avalonia.Controls/Platform/ScreenHelper.cs:14]: I think we need to add a `Rect.ContainsExclusive` method or something, as IMO the `Rect.Contains` logic is just plain confusing (it's inherited from WPF).

File: src/Avalonia.Controls/Platform/IScreenImpl.cs [hunk 3]
```diff
@@ -1,11 +1,19 @@
...
 {
     public interface IScreenImpl
     {
         int ScreenCount { get; }
 
         IReadOnlyList<Screen> AllScreens { get; }
+
+        Screen? ScreenFromWindow(IWindowBaseImpl window);
+
+        Screen? ScreenFromPoint(PixelPoint point);
```
Comment (kekekeks) [inline src/Avalonia.Controls/Platform/IScreenImpl.cs:15]: Can't be implemented with Wayland

File: src/Avalonia.Controls/Platform/IScreenImpl.cs [hunk 4]
```diff
@@ -1,11 +1,19 @@
...
+
 namespace Avalonia.Platform
 {
     public interface IScreenImpl
     {
         int ScreenCount { get; }
 
         IReadOnlyList<Screen> AllScreens { get; }
+
+        Screen? ScreenFromWindow(IWindowBaseImpl window);
```
Comment (kekekeks) [inline src/Avalonia.Controls/Platform/IScreenImpl.cs:13]: Which Screen should we return if window intersects multiple ones?

Comment (maxkatz6) [reply src/Avalonia.Controls/Platform/IScreenImpl.cs:13]: Isn't it something for OS to decide?

File: src/Avalonia.Controls/Platform/IScreenImpl.cs (hunk 3 above)
Comment (grokys) [reply src/Avalonia.Controls/Platform/IScreenImpl.cs:15]: Then it should return `null` i guess?

Comment (danwalmsley) [reply src/Avalonia.Controls/Platform/IScreenImpl.cs:15]: public Screen ScreenFromPoint(PixelPoint point)
        {
            return ScreenHelper.ScreenFromPoint(point, AllScreens);
        }

Comment (danwalmsley) [reply src/Avalonia.Controls/Platform/IScreenImpl.cs:15]: even with this?

File: src/Avalonia.Controls/Platform/ScreenHelper.cs (hunk 2 above)
Comment (danwalmsley) [reply src/Avalonia.Controls/Platform/ScreenHelper.cs]: fixed

**Comment by jmacato** [comment]
Closing due to inactivity and merge conflict

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0019242-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by danwalmsley — APPROVED** [review]

**Pull request merged**